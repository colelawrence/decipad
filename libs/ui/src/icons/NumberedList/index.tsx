import { FC } from 'react';
import { cssVar } from '../../primitives';

export const NumberedList = (): ReturnType<FC> => (
  <svg viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
    <title>NumberedList</title>
    <path
      d="M7.5 11H13.5"
      stroke={cssVar('iconColorHeavy')}
      stroke-width="1.3"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
    <line
      x1="7.65"
      y1="4.35"
      x2="13.55"
      y2="4.35"
      stroke={cssVar('iconColorHeavy')}
      stroke-width="1.3"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
    <path
      d="M2.2 3.67779L3.46666 2.20001V6"
      stroke={cssVar('iconColorHeavy')}
      stroke-width="0.886664"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
    <path
      d="M2.54546 9.95294C2.54546 9.71471 2.74182 9 3.52727 9C4.31273 9 4.50909 9.47647 4.50909 9.95294C4.50909 10.4294 4.06787 11.0959 3.52727 11.6206C2.98668 12.1453 2.46364 12.8118 2.3 13.05H5"
      stroke={cssVar('iconColorHeavy')}
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="0.944999"
    />
  </svg>
);
