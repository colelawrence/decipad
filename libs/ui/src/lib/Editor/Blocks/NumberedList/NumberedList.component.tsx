import { OrderedList } from '@chakra-ui/react';
import React from 'react';
import { RenderElementProps } from 'slate-react';

export const NumberedList = ({
  attributes,
  children,
}: RenderElementProps): JSX.Element => {
  return <OrderedList {...attributes}>{children}</OrderedList>;
};
