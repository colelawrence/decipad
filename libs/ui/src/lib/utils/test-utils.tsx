import { ChakraProvider } from '@chakra-ui/react';
import { render, RenderOptions } from '@testing-library/react';
import React from 'react';
import { theme } from '../theme';

const AllTheProviders: React.FC = ({ children }) => {
  return (
    <ChakraProvider resetCSS theme={theme}>
      {children}
    </ChakraProvider>
  );
};

const customRender = (
  ui: JSX.Element,
  options?: Omit<RenderOptions, 'queries'>
) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';
// override render method
export { customRender as render };
