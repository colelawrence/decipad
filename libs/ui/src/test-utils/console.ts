export const mockConsoleWarn = () => {
  const originalConsoleWarn = console.warn;
  const mockConsoleWarn: jest.MockedFunction<typeof console.warn> = jest.fn();

  beforeEach(() => {
    console.warn = jest.fn();
    mockConsoleWarn.mockClear();
  });
  afterEach(() => {
    console.warn = originalConsoleWarn;
  });

  return mockConsoleWarn;
};
