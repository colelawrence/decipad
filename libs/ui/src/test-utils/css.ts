import cssVarsPonyfill from 'css-vars-ponyfill';

export const findParentWithStyle = <P extends keyof CSSStyleDeclaration>(
  element: Element | null,
  propertyName: P
):
  | (Pick<CSSStyleDeclaration, P> & {
      element: Element;
      styles: CSSStyleDeclaration;
    })
  | null => {
  if (element === null) {
    return null;
  }
  const styles = getComputedStyle(element);
  if (!styles[propertyName]) {
    return findParentWithStyle(element.parentElement, propertyName);
  }
  return {
    [propertyName]: styles[propertyName],
    element,
    styles,
    // cannot type dynamic property key based on type parameter
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
  } as any;
};

export const applyCssVars = (): Promise<() => void> => {
  return new Promise((resolve, reject) => {
    const cleanup = () =>
      [...document.head.querySelectorAll('[data-cssvars="out"]')].forEach(
        (element) => element.remove()
      );
    cssVarsPonyfill({
      onError: reject,
      onComplete: () => resolve(cleanup),
    });
  });
};

export const applyPrefersColorScheme = async (
  colorScheme: 'dark' | 'light'
): Promise<() => void> => {
  // JSDOM ignores CSSOM modifications, so build a CSS string and create and append a style element instead
  const styleText = [...document.styleSheets]
    .flatMap((styleSheet) => [...styleSheet.cssRules])
    .flatMap((rule) => {
      if (rule.type === CSSRule.MEDIA_RULE) {
        const mediaRule = rule as CSSMediaRule;
        for (
          let mediaIndex = 0;
          mediaIndex < mediaRule.media.length;
          mediaIndex += 1
        ) {
          if (
            mediaRule.media[mediaIndex].replace(/\s/g, '') ===
            `(prefers-color-scheme:${colorScheme})`
          ) {
            return [...mediaRule.cssRules];
          }
        }
      }
      return [];
    })
    .flatMap((rule) => rule.cssText)
    .join(' ');

  const styleElement = document.createElement('style');
  styleElement.innerHTML = styleText;
  document.head.appendChild(styleElement);

  const cleanupCssVars = await applyCssVars();

  return () => {
    cleanupCssVars();
    styleElement.remove();
  };
};
