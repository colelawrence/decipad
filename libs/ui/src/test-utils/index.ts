export * from './menu';
export * from './language';
