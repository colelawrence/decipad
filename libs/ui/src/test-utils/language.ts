import {
  Result,
  runCode as run,
  SerializedTypeKind,
  serializeResult,
} from '@decipad/computer';

export async function runCode<T extends SerializedTypeKind>(
  code: string
): Promise<Result.Result<T>> {
  const result = await run(code);
  return serializeResult<T>(result.type, result.value);
}
