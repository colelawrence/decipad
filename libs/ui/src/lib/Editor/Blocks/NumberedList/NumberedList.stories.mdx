import { Meta, Story, Canvas, Props } from '@storybook/addon-docs/blocks';
import { NumberedList } from './NumberedList.component';
import { ListItem } from '@chakra-ui/layout';

<Meta
  title="Editor/Blocks/Lists/NumberedList"
  parameters={{
    docs: {
      source: {
        code: `<NumberedList
  element={{
    children: [],
  }}
  attributes={{ 'data-slate-node': 'element', ref: null }}
>
  <ListItem>Hello World</ListItem>
</NumberedList>`,
      },
    },
  }}
  argTypes={{
    children: {
      defaultValue: 'Hello World',
      description: 'The list items of the component',
      type: 'string',
    },
  }}
  component={NumberedList}
/>

# NumberedList

The NumberedList component is just a basic chakra ui `ol` component that has been adapted to be a slate block element component.

export const Template = (args) => (
  <NumberedList {...args}>
    <ListItem>{args.children}</ListItem>
    <ListItem>{args.children}</ListItem>
    <ListItem>{args.children}</ListItem>
  </NumberedList>
);

<Canvas>
  <Story name="NumberedList">{Template.bind({})}</Story>
</Canvas>

<Props story="NumberedList" />
