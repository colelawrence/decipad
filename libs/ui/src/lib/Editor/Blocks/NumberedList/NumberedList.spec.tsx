import { ListItem } from '@chakra-ui/layout';
import React from 'react';
import { render } from 'test-utils';
import { NumberedList } from './NumberedList.component';

describe('NumberedList Block', () => {
  it('renders', () => {
    const { container, getByText } = render(
      <NumberedList
        element={{ children: [] }}
        attributes={{ 'data-slate-node': 'element', ref: null }}
      >
        <ListItem>Hello World</ListItem>
      </NumberedList>
    );

    expect(container).toBeDefined();
    getByText('Hello World');
  });

  it('matches snapshot', () => {
    const { container } = render(
      <NumberedList
        element={{ children: [] }}
        attributes={{ 'data-slate-node': 'element', ref: null }}
      >
        <ListItem>Hello World</ListItem>
      </NumberedList>
    );

    expect(container).toMatchSnapshot();
  });
});
