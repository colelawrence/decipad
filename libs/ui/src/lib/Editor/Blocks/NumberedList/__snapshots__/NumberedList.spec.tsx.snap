// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberedList Block matches snapshot 1`] = `
.emotion-0 {
  list-style-type: decimal;
  -webkit-margin-start: 1em;
  margin-inline-start: 1em;
}

<div>
  <ol
    class="emotion-0"
    data-slate-node="element"
    role="list"
  >
    <li
      class="emotion-1"
    >
      Hello World
    </li>
  </ol>
</div>
`;
